include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -45,32 +64,54 @@ namespace sparrow
         explicit arrow_proxy(ArrowArray&& array, ArrowSchema&& schema);
         /// Constructs an arrow_proxy which takes the ownership of the ArrowArray and uses the provided
         /// ArrowSchema. The array is released when the arrow_proxy is destroyed. The schema is not released.
-        explicit arrow_proxy(ArrowArray&& array, ArrowSchema* schema);
+        explicit arrow_proxy(ArrowArray&& array, ArrowSchema* schema, ownership schema_ownership);
         /// Constructs an arrow_proxy which uses the provided ArrowArray and ArrowSchema.
         /// Neither the array nor the schema are released when the arrow_proxy is destroyed.
-        explicit arrow_proxy(ArrowArray* array, ArrowSchema* schema);
-
-        ~arrow_proxy();
+        explicit arrow_proxy(
```

Comment:
It should be one [`arrow_data_ownership`](https://github.com/man-group/sparrow/blob/main/include/sparrow/c_interface.hpp#L93) which lets the user specify both ownerships.
Similar to [`external_array_data`](https://github.com/man-group/sparrow/blob/main/include/sparrow/array/external_array_data.hpp#L120), also see [how it looks when the user specifies it](https://github.com/man-group/sparrow/blob/11ef0540b3227b79cf66fb0f331f493ff488b6de/test/external_array_data_creation.hpp#L176-L195).

Reply:
Ok done

Reply:
removed finally: https://github.com/man-group/sparrow/pull/206#discussion_r1758423297